tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -113,16 +115,15 @@ ctorForProductType constructorName typeParameterName children fields = ctorForTy
   toTypeChild (MkChildren field) = ("extra_children", toType field)
 
 -- | Build Q Constructor for leaf types (nodes with no fields or subtypes)
-ctorForLeafType :: Named -> DatatypeName -> Name -> Q Con
-ctorForLeafType named datatypeName typeParameterName =
-  case (named, datatypeName) of
-    (Anonymous, DatatypeName name) -> ctorForTypes Anonymous name [annotation]
-    (Named, DatatypeName name) -> ctorForTypes Named name [annotation, ("bytes", conT ''Text)]
-  where annotation = ("ann", varT typeParameterName) -- ann :: a
+ctorForLeafType :: DatatypeName -> Name -> Q Con
+ctorForLeafType (DatatypeName name) typeParameterName = ctorForTypes name
+  [ ("ann",   varT typeParameterName) -- ann :: a
+  , ("bytes", conT ''Text)
+  ]
 
 -- | Build Q Constructor for records
-ctorForTypes :: Named -> String -> [(String, Q TH.Type)] -> Q Con
-ctorForTypes named constructorName types = recC (toName named constructorName) recordFields where
+ctorForTypes :: String -> [(String, Q TH.Type)] -> Q Con
```

Comment:
We no longer generate constructors for `Anonymous` nodes, so we don’t need to pass the `Named` parameter around; we can just assume the constructor name will be `Named`.